### DISCUSSION TITLE
Health Checks of the ServiceActivator

### DISCUSSION THREAD

**Discussion by preardon** [BrighterCommand/Brighter#1539] [Q&A]
I am implementing Brighter worker process and am looking for the best way to Check the Health of the process. I have Added AspNet.Core so that I can use the usual HTTP Health Check (DB Connection, Bus Connection ect.) but I wanted a bit of insight into the ServiceActivator itself and have come up with the following

`
    public class BrighterHealthCheck : IHealthCheck
    {
        private readonly IDispatcher _dispatcher;

        public BrighterHealthCheck(IDispatcher dispatcher)
        {
            _dispatcher = dispatcher;
        }

        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = new CancellationToken())
        {
            var unhealthyConsumers = _dispatcher.Consumers.Where(c => c.State != ConsumerState.Open);

            if (unhealthyConsumers.Any())
            {
                var status = HealthStatus.Degraded;
                if (unhealthyConsumers.Count() == _dispatcher.Consumers.Count())
                {
                    status = HealthStatus.Unhealthy;
                }

                return new HealthCheckResult(status,
                    $"{unhealthyConsumers.Count()} UnHealthy Consumers From {_dispatcher.Consumers.Count()} on Host {_dispatcher.HostName}");
            }

            return HealthCheckResult.Healthy($"{_dispatcher.HostName} is Healthy");
        }
    }
`

What are your thoughts and do you think there is any more that I should add?

**Comment by iancooper** [comment]
@holytshirt Useful thread for any discussions about how we do better telemetry

**Reply by preardon** [reply]
[Paramore.Brighter.ServiceActivator.Extensions.Diagnostics](https://github.com/BrighterCommand/Brighter/tree/master/src/Paramore.Brighter.ServiceActivator.Extensions.Diagnostics) now has an example of this
